internal/dao/tests/reservation_test.go

```diff
@@ -348,3 +354,42 @@ func TestReservationFinish(t *testing.T) {
 		require.ErrorIs(t, err, dao.ErrAffectedMismatch)
 	})
 }
+
+func TestReservationRate(t *testing.T) {
+	reservationDao, ctx := setupReservation(t)
+
+	t.Run("errors out on two fast consecutive reservation on account", func(t *testing.T) {
+		defer reset()
+		res := newNoopReservation()
+		err := reservationDao.CreateNoop(ctx, res)
+		require.NoError(t, err)
+
+		for i := 0; i < 4; i++ {
+			res = newNoopReservation()
+			err = reservationDao.CreateNoop(ctx, res)
+			require.NoError(t, err)
+		}
+
+		res = newNoopReservation()
+		err = reservationDao.CreateNoop(ctx, res)
+		require.ErrorContains(t, err, "failed to create reservation record")
+	})
+
+	reservationDao, ctx = setupReservationOrg2(t)
+
+	t.Run("does not error out for quick reservations from different account", func(t *testing.T) {
+		defer reset()
+		res := newNoopReservation()
+		err := reservationDao.CreateNoop(ctx, res)
+		require.NoError(t, err)
+
+		for i := 0; i < 4; i++ {
+			res = newNoopReservation()
+			res.Reservation.AccountID = 2
```

Comment:
This will be IMO ignored, we need to change the context instead

Reply:
Oh this is a leftover, it is done above.